Library/Homebrew/compilers.rb

```diff
@@ -111,6 +110,14 @@ def find_compiler
           version = versions.non_apple_gcc_version(name)
           yield Compiler.new(name, version) if version
         end
+      when :clang
+        # silently substitute clang to clang-omp when openmp is required.
+        version = if @formula.needs_compiler_feature?(:openmp)
+          versions.clang_omp_build_version
```

Comment:
There is a flaw here. The version of clang-omp is the version of llvm it based on. Therefore it cannot be compared with the system clang build version.